check-dts/map.errors.ts

```diff
@@ -0,0 +1,83 @@
+import { mapper } from '../src/map';
+import type { Rgb } from '../src/rgb/types';
+import type { Hsl } from '../src/hsl/types';
+import type { Okhsv } from '../src/okhsv/types';
+
+// ===
+// mode = `undefined`
+
+const doubler = mapper(v => {
+	return v * 2;
+});
+
+// THROWS Type 'Rgb' is missing the following properties from type 'Hsl': h, s, l
+const case_3_expect_error: Hsl = doubler({
```

Comment:
@ai I added tests for `.d.ts` files.

Reply:
I use this typing test system in all my open source.

For instance, in PostCSS https://github.com/postcss/postcss/blob/main/test/errors.ts